### PR TITLE
Add main governance document formalizing project leadership structure

### PR DESCRIPTION
**Pull request by CAM-Gerlach**
This PR adds the proposed main Governance Document for the Spyder project, along with a supporting list of its relevant People (to be revised as necessary). As described at some length therein, it is directly based on the [Pandas Main Governance Document](https://github.com/pandas-dev/pandas-governance/blob/f7352be07872048a07be3d3614ae142e004568df/governance.md) that was given as an exemplar by NumFOCUS of what was required, which is a fairly good fit for as, as it relatively closely matches our existing governance model. I then modified it as required to eliminate sections and provisions that aren't relevant for Spyder, adding in various bits that are, modifying it as needed to reflect the differences between the two projects, clarifying some segments left ambiguous in the original document, and addressing some editorial issues.

Also, I've added (also modeled closed on Pandas) a corresponding People List, listing the people in various positions relevant to the Governance Document, which will serve as a public record of such as required by NumFOCUS's transparency requirements and also list the required five NumFOCUS contacts, among others. In some places the assignments of people to roles are sketchy or incomplete; we can discuss this further at the meeting and make the necessary adjustments as needed.

Hopefully we can all try to look it over before then, but we can kick off the formal discussion with the team at our weekly developer meeting, and go from there. We'll need to have this voted on, approved and merged before we can start the process to become a NumFOCUS Fiscally Sponsored project, which we can talk about at the meeting as well, but we should also be sure we are all happy with it and any feedback has been incorporated.

### CODE REVIEW

File: governance.md [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+Users must be Core Developers to be granted Write-level access and above on these repositories, as well as Admin-level access on other repositories in the Spyder-IDE organization.
+It is the Core Developers that vote on specific, major, high-level project governance issues as specified in this document.
+Additionally, the Core Developers have primary responsibility for guiding the code review process and merging pull requests, and for administering and maintaining any key services and infrastructure on which the project relies.
+
+To become eligible for becoming a Core Developer, an individual must be a Project Contributor who has produced contributions that are substantial in quality and quantity, and sustained over at least one year.
+Potential Core Developers are nominated by an Core Developers and voted upon by the Core Developers after asking if the potential Core Developer is interested and willing to serve in that capacity.
```
Comment (jitseniesen) [inline governance.md]:
```suggestion
Potential Core Developers are nominated by a Core Developer and voted upon by the Core Developers after asking if the potential Core Developer is interested and willing to serve in that capacity.
```

**Review by jitseniesen — COMMENTED** [review]
Looks fine at first sight.

File: governance.md (hunk 1 above)
Comment (CAM-Gerlach) [reply governance.md]: Thanks, I fixed that and revised the sentence a bit to make it shorter, add a missing comma and reduce repetition.

**Comment by CAM-Gerlach** [comment]
Hey @jitseniesen , do you want to be added to the NumFOCUS and/or CoC teams in the People list? I would have, but I wasn't sure how active you were. BTW, are you able to make the next developer meeting this Thursday at 20:00 UTC on the Spyder Zoom?

**Comment by jitseniesen** [comment]
> Hey @jitseniesen , do you want to be added to the NumFOCUS and/or CoC teams in the People list? I would have, but I wasn't sure how active you were. BTW, are you able to make the next developer meeting this Thursday at 20:00 UTC on the Spyder Zoom?

I am planning to attend the meeting, we can have a chat about it then.

File: governance.md [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+Potential Core Developers are nominated and voted upon by the existing Core Developers, after asking if the candidate is interested and willing to serve in that capacity.
+A majority of all active Core Developers must vote in favor of adding the new member, unless at least one core developer votes against, in which case at least two thirds of all active Core Developers must vote in favor.
+The Core Developers will be initially formed from the set of existing Contributors who are members of the Core Developers Team as of July 2022.
+
+When considering potential new members, the Core Developers will look at candidates with a comprehensive view of their contributions.
+This will include but is not limited to code, code review, infrastructure work, participation in communications channels, community help/building, education and outreach, design work, etc.
+The Project want to encourage a diverse array of backgrounds, viewpoints and talents on the Core Developer team, which is why code is not defined as the sole metric on which Core Developers membership will be evaluated.
```
Comment (ccordoba12) [inline governance.md]:
```suggestion
The Project wants to encourage a diverse array of backgrounds, viewpoints and talents on the Core Developer team, which is why code is not defined as the sole metric on which Core Developers membership will be evaluated.
```

File: governance.md [hunk 3]
```diff
@@ -0,0 +1,258 @@
...
+
+If a Core Developer falls well below the threshold of activity over a period of one year that would qualify a candidate to be considered for Core Developer membership, they will be considered inactive.
+The inactive Core Developer will then be approached by the Lead Maintainer to see if they plan on returning to active participation.
+If they respond in the negative, or do not respond within 30 days, they will be removed as a Core Developer after a majority vote by the active Core Developers.
+If they plan on returning to active participation soon, they will be given a grace period of six months.
+If they do not do so after that time, they will be removed after a majority vote by the active Core Developers.
+
+Upon removal, such Core Developers will remain members of the Spyder-IDE organization, and be removed from any Teams they may be members of and added to the Former Developers Team.
+Retired Core Developers members will be also included on the People list, acknowledging the period during which they were active in the Core Developers.
```
Comment (ccordoba12) [inline governance.md]:
```suggestion
Retired Core Developers members will be also included on the People list, acknowledging the period during which they were active in the Core Developers Team.
```

File: governance.md [hunk 4]
```diff
@@ -0,0 +1,258 @@
...
+* During any period for which there are fewer than three individuals holding the role of Organization Owner, the Governance and Guidelines repository maintainer shall be appointed a temporary, acting Organization Owner for the specific purposes of facilitating the transition, reducing bus factor, maintaining a quorum and mediating any disagreements between the remaining Owners; and will be immediately removed upon confirmation of a new Owner.
+
+Additionally, Organization Owners have by default the "Admin" role on all Spyder-IDE Organization repositories, including `spyder` and `spyder-kernels`.
+For security reasons, these should be the only individuals with the "Admin" role on the `spyder` and `spyder-kernels` repositories.
+
+
+### Organization Members
+
+Organization Members are Contributors who, due to a reasonable and appropriate need (such having a formal relationship with the Spyder Project, or being considered or mentored for eventual nomination as a Core Developer), are added as Members of the Spyder-IDE Organization.
```
Comment (ccordoba12) [inline governance.md]:
```suggestion
Organization Members are Contributors who, due to a reasonable and appropriate need (such as having a formal relationship with the Spyder Project, or being considered or mentored for eventual nomination as a Core Developer), are added as Members of the Spyder-IDE Organization.
```

File: governance.md [hunk 5]
```diff
@@ -0,0 +1,258 @@
...
+Such Organization Members (other than Core Developers) are added by the Lead Maintainer, with the advice and consent of the Core Developers.
+They may be granted permissions of "Triage" or lower on the `spyder` or `spyder-kernels` repositories, or "Maintain" or lower on any other repository, and be added to any Team with the same.
```
Comment (ccordoba12) [inline governance.md]:
```suggestion
They may be granted permissions of "Triage" or lower on the `spyder` or `spyder-kernels` repositories, or "Maintain" or lower on any other repository, and be added to any Team with the same permissions.
```

**Review by mrclary — DISMISSED** [review]

File: governance.md [hunk 6]
```diff
@@ -0,0 +1,258 @@
...
+
+
+
+## The Project
+
+The Spyder Project (the Project) is an open source software project affiliated with the 501(c)3 NumFOCUS Foundation.
+The goal of the Project is to develop graphical applications, libraries and utilities for scientific computing, data exploration/analysis/visualization, and research software development for the Python programming language.
+The software developed by the Project (the Project Software) is released under the [MIT](https://opensource.org/licenses/MIT) (or similar) open source license, developed openly and hosted in public GitHub repositories under the [Spyder-IDE GitHub Organization](https://github.com/spyder-ide).
+Examples of Project Software include the main [Spyder](https://github.com/spyder-ide/spyder) scientific Python development environment, the [Spyder-Kernels](https://github.com/spyder-ide/spyder-kernels) support package, the [QtPy](https://github.com/spyder-ide/spyder) Qt abstraction layer, and the [Spyder-Docs](https://github.com/spyder-ide/spyder-docs) documentation.
```
Comment (dalthviz) [inline governance.md]: Fix link to QtPy:
```suggestion
Examples of Project Software include the main [Spyder](https://github.com/spyder-ide/spyder) scientific Python development environment, the [Spyder-Kernels](https://github.com/spyder-ide/spyder-kernels) support package, the [QtPy](https://github.com/spyder-ide/qtpy) Qt abstraction layer, and the [Spyder-Docs](https://github.com/spyder-ide/spyder-docs) documentation.
```

File: governance.md [hunk 7]
```diff
@@ -0,0 +1,258 @@
...
+This section describes the governance and leadership model of the Project.
+
+The foundations of Project governance are:
+
+* Openness and Transparency
+* Active Contribution
+* Institutional Neutrality
+
+Traditionally, Project leadership has been provided by a Lead Maintainer (Carlos Cordoba, and previously Spyder creator Pierre Raybaut) and a subset of Contributors, called the Spyder Core Developers, whose active and consistent contributions to the Project have been recognized by their being added to the Core Developers GitHub Team, which grants the Write role on the primary Project GitHub repository.
+In general, all Project decisions are made through consensus among the Core Developers with input from the Community.
```
Comment (dalthviz) [inline governance.md]: Just to settle that the Core Developers are part of the Community:
```suggestion
In general, all Project decisions are made through consensus among the Core Developers with input from the rest off the Community.
```

File: governance.md [hunk 8]
```diff
@@ -0,0 +1,258 @@
...
+Example Teams may include, but are not limited to, Documentation, Website, and User Experience.
+Additionally, the Junior Developers Team is reserved for Organization Members who are not Core Developers, and generally intended for those that are not otherwise serving in a highly specialized role (such as a UX designer, technical writer, and the like).
+
+
+
+## Transparency
+
+Unless specifically required, all Core Developer and Organization discussions and activities should be public and done in collaboration and discussion with the Project Contributors and Community.
+The Core Developers will have a private Gitter channel that will be used sparingly when a specific matter requires privacy.
+When private communications and decisions are needed, the Core Developers will do its best to summarize those to the Community after eliding personal/private/sensitive information that should not be posted to the public internet.
```
Comment (dalthviz) [inline governance.md]:
```suggestion
When private communications and decisions are needed, the Core Developers will do its best to summarize those to the rest of the Community after eliding personal/private/sensitive information that should not be posted to the public internet.
```

File: governance.md [hunk 9]
```diff
@@ -0,0 +1,258 @@
...
+The Lead Maintainer, consulting with the Core Developers, can create other Teams that provide leadership, guidance and coordination for specific aspects of the project.
+While Team Maintainers/Leads must be Core Developers, Team members may be any Member of the Organization.
+Example Teams may include, but are not limited to, Documentation, Website, and User Experience.
+Additionally, the Junior Developers Team is reserved for Organization Members who are not Core Developers, and generally intended for those that are not otherwise serving in a highly specialized role (such as a UX designer, technical writer, and the like).
+
+
+
+## Transparency
+
+Unless specifically required, all Core Developer and Organization discussions and activities should be public and done in collaboration and discussion with the Project Contributors and Community.
```
Comment (dalthviz) [inline governance.md]:
```suggestion
Unless specifically required, all Core Developer and Organization discussions and activities should be public and done in collaboration and discussion with the Project Contributors and Community in general.
```

**Review by dalthviz — COMMENTED** [review]
Hi! Left some suggestions (fixing a link and some phrasing). Otherwise this LGTM :+1:

File: governance.md (hunk 7 above)
Comment (CAM-Gerlach) [reply governance.md]:
```suggestion
In general, all Project decisions are made through consensus among the Core Developers with input from the rest of the Community.
```

Good point—I just fixed a typo in the suggestion ("of", not "off"; you got it right the other spot)

**Review by dalthviz — DISMISSED** [review]
:+1:

**Comment by jitseniesen** [comment]
> Hey @jitseniesen , do you want to be added to the NumFOCUS and/or CoC teams in the People list?

I am happy to be added to the NumFOCUS team.

File: people.md [hunk 10]
```diff
@@ -0,0 +1,82 @@
...
+* Daniel Althviz Moré ([@dalthviz](https://github.com/dalthviz))
+* Pierre Raybaut ([@PierreRaybaut](https://github.com/PierreRaybaut))
+
+
+## NumFOCUS Representatives
+
+* Carlos Cordoba ([@ccordoba12](https://github.com/ccordoba12))
+* C.A.M. Gerlach ([@CAM-Gerlach](https://github.com/CAM-Gerlach))
+* Daniel Althviz Moré ([@dalthviz](https://github.com/dalthviz))
+* Quentin Peter ([@impact27](https://github.com/impact27))
```
Comment (CAM-Gerlach) [inline people.md]:
```suggestion
* Jitse Niesen ([@jitseniesen](https://github.com/jitseniesen))
```

**Review by dalthviz — DISMISSED** [review]

**Review by mrclary — DISMISSED** [review]

File: governance.md [hunk 11]
```diff
@@ -0,0 +1,258 @@
...
+
+The Project is developed by a global collective of distributed developers, called Contributors.
+Contributors are individuals who have contributed code, documentation, designs or other work to one or more Project repositories.
+Anyone can be a Contributor.
+Contributors can be affiliated with any legal entity or none.
+Contributors participate in the project by submitting, reviewing and discussing GitHub Pull Requests and Issues and participating in open and public Project discussions on GitHub, community calls and elsewhere.
+The foundation of Project participation is openness and transparency.
+A list of all Contributors can be found on the "Contributors" page of each individual Project repository, such as the [one for the main Spyder repository](https://github.com/spyder-ide/spyder/graphs/contributors).
+
+Contributors with a sustained, formal relationship of the project are added as Organization Members, part of the Spyder-IDE Organization.
```
Comment (ccordoba12) [inline governance.md]:
```suggestion
Contributors with a sustained, formal relationship to the project are added as Organization Members, part of the Spyder-IDE Organization.
```

Seems like a typo?

File: governance.md [hunk 12]
```diff
@@ -0,0 +1,258 @@
...
+
+* If a new Lead Maintainer or designated successor is successfully confirmed by vote, they are added as a new Organization Owner.
+* If one of the individuals whose position grants them the role of Organization Owner is recalled or voluntarily retires from their respective position, they are obligated to immediately remove themselves, or be removed, from this role.
+* If a vote is called on the removal of an Organization Owner or a recall of a Organization Member whose position grants them this privilege, upon concurrence of two of the three Organization Owners, the Member shall temporarily be removed for the duration of the vote or two weeks, whichever is less, and then restored unless a two thirds majority is reached.
+* If an Owner or their account are believed to physically or mentally compromised, either on their request or with the concurrence of the other two Owners, they may be temporarily removed for a period of up to two weeks and then restored, unless two thirds of the remaining active Core Developers vote for their permanent removal.
```
Comment (ccordoba12) [inline governance.md]:
```suggestion
* If an Owner or their account are believed to be physically or mentally compromised, either on their request or with the concurrence of the other two Owners, they may be temporarily removed for a period of up to two weeks and then restored, unless two thirds of the remaining active Core Developers vote for their permanent removal.
```

File: governance.md [hunk 13]
```diff
@@ -0,0 +1,258 @@
...
+While Team Maintainers/Leads must be Core Developers, Team members may be any Member of the Organization.
+Example Teams may include, but are not limited to, Documentation, Website, and User Experience.
+Additionally, the Junior Developers Team is reserved for Organization Members who are not Core Developers, and generally intended for those that are not otherwise serving in a highly specialized role (such as a UX designer, technical writer, and the like).
+
+
+
+## Transparency
+
+Unless specifically required, all Core Developer and Organization discussions and activities should be public and done in collaboration and discussion with the Project Contributors and Community in general.
+The Core Developers will have a private Gitter channel that will be used sparingly when a specific matter requires privacy.
```
Comment (ccordoba12) [inline governance.md]:
```suggestion
The Core Developers will have a private Discord channel that will be used sparingly when a specific matter requires privacy.
```

File: governance.md [hunk 14]
```diff
@@ -0,0 +1,258 @@
...
+An institution funding at least one Spyder Contributor.
+
+* Individual contributions acknowledged in the Releases page, Changelog, commit messages and Pull Requests.
+* Ability to influence the project through the participation of their Contributor.
+
+
+
+## Breach
+
+Non-conformance with the terms of the governance documents shall be reported to the Core Developers, either through public (issue on the Governance and Guidelines repository) or private (Core Developer Gitter) channels as deemed appropriate.
```
Comment (ccordoba12) [inline governance.md]:
```suggestion
Non-conformance with the terms of the governance documents shall be reported to the Core Developers, either through public (issue on the Governance and Guidelines repository) or private (Core Developer Discord) channels as deemed appropriate.
```

**Review by ccordoba12 — COMMENTED** [review]
Thanks a lot @CAM-Gerlach for your work on this and all other people that participated with your reviews and comments in this PR.

This looks almost ready to me, except for the very minor suggestions I left below.

File: CHANGELOG.md [hunk 15]
```diff
@@ -3,6 +3,12 @@
 All substantive changes to the normative governance, guideline and policy documents will be documented here.
 
 
+## 2022-08-11
```
Comment (CAM-Gerlach) [inline CHANGELOG.md]:
```suggestion
## 2022-10-13
```

Set final planned merge date

**Comment by CAM-Gerlach** [comment]
@PierreRaybaut if you'd like to give feedback on this, please feel free as well. I figured I'd ping you since you were Spyder's original creator, and your name is also mentioned a couple times in here. Thanks!

File: governance.md [hunk 16]
```diff
@@ -0,0 +1,258 @@
...
+
+Organization Members are Contributors who, due to a reasonable and appropriate need (such as having a formal relationship with the Spyder Project, or being considered or mentored for eventual nomination as a Core Developer), are added as Members of the Spyder-IDE Organization.
+Such Organization Members (other than Core Developers) are added by the Lead Maintainer, with the advice and consent of the Core Developers.
+They may be granted permissions of "Triage" or lower on the `spyder` or `spyder-kernels` repositories, or "Maintain" or lower on any other repository, and be added to any Team with the same permissions.
+Otherwise, they retain the same rights and privileges as other Contributors, including the possibility of earning membership in the Core Developers through their sustained contributions to the Project.
+
+
+### Repository Collaborators
+
+Per-repository Collaborators are Contributors who, thanks to their record of contributions to a specific repository, to further deeper collaboration, or a specific need for their expertise, are given a Role on a particular Spyder-IDE Organization repository.
```
Comment (mrclary) [inline governance.md]:
```suggestion
Per-repository Collaborators are Contributors who, thanks to their record of contributions to a specific repository, to further deepen collaboration, or a specific need for their expertise, are given a Role on a particular Spyder-IDE Organization repository.
```

File: governance.md [hunk 17]
```diff
@@ -0,0 +1,258 @@
...
+They can be added at the discretion of the Core Developer maintainers of the other repositories under the Spyder organization (aside from `spyder`/`spyder-kernels`), with the consent of the Spyder Lead Maintainer.
+They may be granted any Role up to Write, or Maintain in special circumstances, for the duration of their collaboration.
+
+
+## Teams
+
+This section concerns the various Teams that may be established of Core Developers and other Organization Members, both those for the purposes of governance, and those for coordination and specialization within the Organization.
```
Comment (mrclary) [inline governance.md]:
```suggestion
This section concerns the various Teams, comprised of Core Developers and other Organization Members, for the purposes of governance and for coordination and specialization within the Organization.
```

**Review by mrclary — CHANGES_REQUESTED** [review]
I left a few grammatical suggestions.

**Review by mrclary — DISMISSED** [review]

**Comment by PierreRaybaut** [comment]
> @PierreRaybaut if you'd like to give feedback on this, please feel free as well. I figured I'd ping you since you were Spyder's original creator, and your name is also mentioned a couple times in here. Thanks!

Hey @CAM-Gerlach , thanks for alerting me to the existence of this PR. I will take a closer look to this document and give feedback.

**Comment by CAM-Gerlach** [comment]
Thanks. Just FYI, we kinda need it to be approved and merged by the time our NumFOCUS application is reviewed (submitted on or about Saturday, Oct 15), which will allow us to be officially supported by the NumFOCUS nonprofit org to help fund the project and ensure it continues to thrive (given recent-ish events), so if you could try to get your feedback in before then (ideally, before or at our regular Thursday afternoon meeting, if you'd like to attend), that would be super :slightly_smiling_face: Especially since everyone needs to approve the final version again if any changes are requested, which can be a bit like herding cats, so the sooner the better.

**Comment by PierreRaybaut** [comment]
> > @PierreRaybaut if you'd like to give feedback on this, please feel free as well. I figured I'd ping you since you were Spyder's original creator, and your name is also mentioned a couple times in here. Thanks!
> 
> Hey @CAM-Gerlach , thanks for alerting me to the existence of this PR. I will take a closer look to this document and give feedback.

@CAM-Gerlach : that is done.

This is a fine document, well done.
(I've just added my humble contribution)

**Comment by CAM-Gerlach** [comment]
Thanks @PierreRaybaut !

**Review by mrclary — APPROVED** [review]

**Review by dalthviz — APPROVED** [review]

**Review by ccordoba12 — APPROVED** [review]
Looks good to me now.

**Review by jitseniesen — APPROVED** [review]

**Review by impact27 — APPROVED** [review]
Looks good

**Review by jitseniesen — APPROVED** [review]

**Review by ccordoba12 — APPROVED** [review]

**Review by dalthviz — APPROVED** [review]

**Comment by CAM-Gerlach** [comment]
Looks like we all approve now, and GitHub has picked it up! We can formally hit the <kbd>Merge</kbd> button in our meeting in 30 minutes time.

**Pull request merged**